doc/osmp_test_architecture.md

```diff
@@ -0,0 +1,224 @@
+# OSMP Test Architecture
+
+The OSMP Test Architecture is designed to test any simulation model complying to the [OSI Sensor Model Packaging (OSMP)](https://github.com/OpenSimulationInterface/osi-sensor-model-packaging) specification.
+Contrary to the S in OSMP, a variety of model typed from environmental effect models over sensor models to traffic participant models can be tested with this architecture.
+The architecture is divided into 3 main test levels based on the credibility assessment level definition by Ahmann et al. [[1](#Ahmann2022)].
+Because the test architecture is specifically designed for OpenMSL open source models, a license check is needed.
+The license check is preceding the 3 main credibility assessment levels as level 0.
+This yields the following credibility assessment level line-up:
+
+- CL 0: [License Check](#cl-0-license-check)
+- CL 1: [Code Verification](#cl-1-code-verification)
+- CL 2: [Qualitative Verification](#cl-2-qualitative-verification)
+- CL 3: [Quantitative Verification](#cl-3-quantitative-verification)
+
+Every credibility assessment level is tested with a separate GitHub action and is further described in the following sections.
+
+<p align="center">
+  <img alt="Credibility Assessment Level" src="./img/credibility_assessment_level.png" width="600"> <br>
+  <em>Discrete credibility assessment level</em>
+</p>
+
+## CL 0: License Check
+
+For credibility assessment level 0 the software licenses are checked.
+A GitHub Action checks every file containing code for an SPDX license identifier.
+The identified licenses are verified against a list of approved open source licenses.
+Approved licenses are: Apache-2.0, BSD-3-Clause, BSD-2-Clause, GPL-2.0, GPL-3.0, LGPL-2.0, LGPL-2.1, LGPL-3.0, MIT, MPL-2.0, CDDL-1.0, EPL-2.0.
```

Comment:
I think we do not have an officially approved list of licenses but I would take out the copy left licenses. Maybe also make a note that the licenses are individually checked by the maintainers regarding their impact.

Reply:
Which licenses exactly would you take out? Isn't the MPL also a copyleft license?

Reply:
Apache-2.0, BSD-3-Clause, BSD-2-Clause, LGPL-2.0, LGPL-2.1, LGPL-3.0, MIT, MPL-2.0

Reply:
Check which ones are used by all the other artifacts

Reply:
I tend to leave out LGPL

Reply:
done.